src/object.ts

```diff
@@ -315,6 +315,57 @@ export class KubernetesObjectApi extends ApisApi {
         return this.requestPromise(localVarRequestOptions);
     }
 
+    /**
+     * List any Kubernetes resources.
+     * @param spec Kubernetes resource spec
+     * @param pretty If \&#39;true\&#39;, then the output is pretty printed.
+     * @param exact Should the export be exact.  Exact export maintains cluster-specific fields like
+     *        \&#39;Namespace\&#39;. Deprecated. Planned for removal in 1.18.
+     * @param exportt Should this value be exported.  Export strips fields that a user can not
+     *        specify. Deprecated. Planned for removal in 1.18.
+     * @param options Optional headers to use in the request.
+     * @return Promise containing the request response and list of [[KubernetesObject]].
+     */
+    public async list(
+        spec: KubernetesObject,
```

Comment:
I don't think we want a `KubernetesObject` here, instead pass in `apiGroup` `apiVersion` & `kind` as well as a `namespaced` parameter.

We'll need to refactor the specUriPath code also.